main.py

```diff
@@ -0,0 +1,233 @@
+        data['predicted_sentiment'] = data['compound'].apply(map_compound_to_sentiment_category2)
+    #Create final csv and save
+    result_data = data[['actual_sentiment', 'predicted_sentiment']]
+    result_data.to_csv(f'./sentiment_results{size}.csv', index=False)
+
+    return result_data
+
+def calculate_precision(actual_sentiment, predicted_sentiment, label: int):
+    TP = sum((actual == label) and (predicted == label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+    FP = sum((actual != label) and (predicted == label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+
+    precision = TP / (TP + FP) if TP + FP != 0 else 0.0
+    return precision
+
+def calculate_recall(actual_sentiment: pd.DataFrame, predicted_sentiment: pd.DataFrame, label: int):
+    TP = sum((actual == label) and (predicted == label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+    FN = sum((actual == label) and (predicted != label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+
+    recall = TP / (TP + FN) if TP + FN != 0 else 0.0
+    return recall
+
+def calculate_accuracy(actual_sentiment: pd.DataFrame, predicted_sentiment: pd.DataFrame):
+    TP = sum((actual == predicted) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+    total_samples = len(actual_sentiment)
+    accuracy = TP / total_samples
+    return accuracy
+
+#Printing Confusion Matrix, Accurancy, Precision, Recall, F1-measure
+def calculate_metrics(size = 3): #size is number of Class Classification. Have to be > 1 and suitable for your data.
+    df = pd.read_csv(f'./sentiment_results{size}.csv')
+
+    actual_sentiment = df['actual_sentiment']
+    predicted_sentiment = df['predicted_sentiment']
+
+    print(f"METRICS FOR {size} CLASSES", end = "\n\n")
+
+    #Confusion matrix
+    if size == 3:
+        unique_labels = [-1, 0, 1]
+        confusion_matrix = np.zeros((3, 3), dtype=int)
+    elif size == 2:
+        unique_labels = [-1, 1]
+        confusion_matrix = np.zeros((2, 2), dtype=int)
+
+    for actual, predicted in zip(actual_sentiment, predicted_sentiment):
+        if actual in unique_labels and predicted in unique_labels:
+            confusion_matrix[unique_labels.index(actual)][unique_labels.index(predicted)] += 1
+
+    print("Confusion Matrix:")
+    print(confusion_matrix)
+
+    #Precision, Recall, Accurancy, F1-measure
+    precision_per_class = {}
+    recall_per_class = {}
+    for label in unique_labels:
+        precision = calculate_precision(actual_sentiment, predicted_sentiment, label)
+        recall = calculate_recall(actual_sentiment, predicted_sentiment, label)
+        precision_per_class[label] = precision
+        recall_per_class[label] = recall
+
+    print("Accurancy:")
+    print(calculate_accuracy(actual_sentiment,predicted_sentiment))
+
+    print("Precision per Class:")
+    print(precision_per_class)
```

Comment:
It is probably easier to follow, when written as 

```suggestion
    print("Precision per Class:", precision_per_class)
```